subsys/net/lib/zzhc/zzhc.c

```diff
@@ -451,6 +454,47 @@ static int check_iccid(struct zzhc *ctx)
 	return rc;
 }
 
+static int check_and_set_dns(struct zzhc *ctx)
+{
+	int err;
+	uint8_t dns_compare[32];
+	struct nrf_in_addr pri_dns;
+	struct nrf_in_addr sec_dns;
+
+	err = nrf_modem_at_cmd(ctx->at_resp, AT_RESPONSE_LEN,
+			       "AT+CGCONTRDP=0");
+
+	if (err) {
+		return -EIO;
+	}
+
+	snprintk(dns_compare, sizeof(dns_compare),
+		 "\"%s\",\"%s\"",
+		 PRIMARY_DNS,
+		 SECONDARY_DNS);
+
+	if (strstr(ctx->at_resp, dns_compare)) {
+		return 0;
+	}
+
+	/* Manually set DNS */
+	err = nrf_inet_pton(NRF_AF_INET, PRIMARY_DNS, &pri_dns);
+	err |= nrf_inet_pton(NRF_AF_INET, SECONDARY_DNS, &sec_dns);
+
+	if (err) {
+		return -EIO;
+	}
```

Comment:
`nrf_inet_pton()` returns 1 on success (at least according to the docs), I don't think this check is correct.